web_query_timeout/controllers/main.py

```diff
@@ -25,3 +25,21 @@ def do_search_read(self, model, fields=False, offset=0, limit=False,
                 'length': 0,
                 'records': 'WebQueryTimeoutException',
             }
+
+    def _call_kw(self, model, method, args, kwargs):
+        """
+        Set the context value that triggers the database query duration when
+        the triggered method is read_group.
+        """
+        if method == 'read_group':
+            kwargs['context']['statement_timeout'] = True
+
+        try:
+            res = super(WebQueryTimeoutDataSet, self)._call_kw(
+                model, method, args, kwargs)
+            return res
+        except QueryCanceledError:
+            return {
+                'length': 0,
+                'records': 'WebQueryTimeoutException',
```

Comment:
You need to add something to anticipate this encoded exception on the javascript side, analogous to https://github.com/EssentNovaTeam/web/blob/cd0aea4ec76c555625dc9a3c1144d6b4b20b84fd/web_query_timeout/static/src/js/web_query_timeout.js#L9. Note that read_group usually returns a list of dictionaries. You might want to adapt to that.